src/TimeoutMigrationTool.ASQ.IntegrationTests/ASQTargetTests.cs

```diff
@@ -0,0 +1,378 @@
+                    Time = new DateTime(2021, 12, 12, 12, 12, 12, DateTimeKind.Utc)
+                },
+                new TimeoutData
+                {
+                    Id = "SomeOtherId",
+                    Headers = new Dictionary<string, string>
+                    {
+                        { "NServiceBus.MessageId", "SomeOtherMessageId" }
+                    },
+                    Destination = "SomeOtherDestination",
+                    State = new byte[2],
+                    Time = new DateTime(2021, 12, 12, 12, 13, 13, DateTimeKind.Utc)
+                },
+            }, 1).ConfigureAwait(true);
+
+            // Assert
+            Assert.ThrowsAsync<Exception>(async () =>
+            {
+                // Act
+                await timeoutTarget.Complete(endpointName).ConfigureAwait(true);
+            });
+        }
+
+        [Test]
+        public async Task Complete_Removes_Staging_Queue_If_Empty()
+        {
+            // Arrange
+            var nameProvider = new DelayedDeliveryTableNameProvider();
+
+            await CreateTimeoutTable(nameProvider.GetDelayedDeliveryTableName(endpointName)).ConfigureAwait(true);
+
+            var timeoutTarget = new ASQTarget(new TestLoggingAdapter(), connectionString, nameProvider);
+            var migrator = await timeoutTarget.PrepareTargetEndpointBatchMigrator(endpointName).ConfigureAwait(true);
+
+            await migrator.StageBatch(new List<TimeoutData>
+            {
+                new TimeoutData
+                {
+                    Id = "SomeID",
+                    Headers = new Dictionary<string, string>
+                    {
+                        { "NServiceBus.MessageId", "SomeMessageId" }
+                    },
+                    Destination = "SomeDestination",
+                    State = new byte[2],
+                    Time = new DateTime(2021, 12, 12, 12, 12, 12, DateTimeKind.Utc)
+                },
+                new TimeoutData
+                {
+                    Id = "SomeOtherId",
+                    Headers = new Dictionary<string, string>
+                    {
+                        { "NServiceBus.MessageId", "SomeOtherMessageId" }
+                    },
+                    Destination = "SomeOtherDestination",
+                    State = new byte[2],
+                    Time = new DateTime(2021, 12, 12, 12, 13, 13, DateTimeKind.Utc)
+                },
+            }, 1).ConfigureAwait(true);
+
+            await migrator.CompleteBatch(1).ConfigureAwait(true);
+
+            // Act
+            await timeoutTarget.Complete(endpointName).ConfigureAwait(true);
+
+            // Assert
+            var stagingTableExists = await DoesTableExist(nameProvider.GetStagingTableName(endpointName)).ConfigureAwait(true);
+
+            Assert.IsFalse(stagingTableExists);
+        }
+
+        async Task DeleteTable(string tableName)
+        {
+            var cloudStorageAccount = CloudStorageAccount.Parse(connectionString);
```

Comment:
It doesn't really matter that much but since we have a dedicated setup and teardown phase would it make sense to create the client once, assign it to a field and then just reuse that?

Reply:
:+1: